extensions/github/src/extension.ts

```diff
@@ -0,0 +1,145 @@
+  constructor(private telemetryLogger: extensionApi.TelemetryLogger) {}
+
+  async getSessions(scopes?: string[]): Promise<readonly extensionApi.AuthenticationSession[]> {
+    if (scopes?.length > 0) throw new Error('Scopes are not supported.');
+    return this.sessions;
+  }
+
+  async createSession(scopes: string[]): Promise<extensionApi.AuthenticationSession> {
+    if (scopes.length > 0) throw new Error('Scopes are not supported.');
+
+    // We only support one session
+    if (this.sessions.length) return this.sessions[0];
+
+    // Get the access token from the preferences
+    const accessToken = getGitHubAccessToken();
+
+    // Check if the token exists
+    if (accessToken === undefined || accessToken.length === 0) {
+      await window.showErrorMessage('No access token provided.');
+      throw new Error('No access token provided.');
+    }
+
+    // Validate the token
+    // (1) Create an Octokit client with the access token
+    // (2) Get the /user endpoint to get user information (username email etc.)
+    const octokit = new Octokit({
+      auth: accessToken,
+    });
+    const { data } = await octokit.users.getAuthenticated();
+
+    // Create a session with our valid access token
+    const session = new GithubSession(
+      `${data.id}`,
+      accessToken,
+      {
+        id: `${data.id}`,
+        label: `${data.name}`,
+      },
+      [],
+    );
+
+    this.telemetryLogger.logUsage('github-create-auth-session');
+
+    // Fire events
+    this._onDidChangeSession.fire({
+      added: [session],
+    });
+
+    // Store the session
+    this.sessions.push(session);
+
+    return session;
+  }
+  async removeSession(sessionId: string): Promise<void> {
+    const removed = this.sessions.filter(session => session.id === sessionId);
+    this._onDidChangeSession.fire({
+      removed: removed,
+    });
+    this.sessions = this.sessions.filter(session => session.id !== sessionId);
+  }
+}
+
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  const telemetryLogger = extensionApi.env.createTelemetryLogger();
+  const authProvider = new GitHubProvider(telemetryLogger);
+
+  // Register the AuthenticationProvider
+  const disposable = extensionApi.authentication.registerAuthenticationProvider('github', 'GitHub', authProvider, {
+    images: {
+      icon: 'icon.png',
+    },
+  });
+  extensionContext.subscriptions.push(disposable);
+  // If we have a defined GitHub preference access token, let's create the session
+  if (getGitHubAccessToken()) {
+    try {
+      await authProvider.createSession([]);
+    } catch (e) {
+      console.error(e);
+      await window.showErrorMessage(`GitHub access token is not valid: ${String(e)}.`);
+    }
+  }
+}
+
+export function deactivate(): void {
+  console.log('stopping github extension');
+}
+
+export function getGitHubAccessToken(): string | undefined {
+  return extensionApi.configuration.getConfiguration('github').get('token');
```

Comment:
A github token should be treated the same way as username/password pair. It cannot be stored in configuration.

Reply:
@benoitf do you have an idea on the way to proceed because I am a bit lost

Reply:
Extension context should have `secrets` property to store secrets and that should be encrypted storage.
`safeStorage` and `safeStorageElectron` in electron that can help with encryption.

Example how to use it from VS Code https://vscode.dev/github/microsoft/vscode/blob/main/src/vs/platform/encryption/electron-main/encryptionMainService.ts#L32